src/AlertsManagement/AlertsManagement/AlertsManagementBaseCmdlet.cs

```diff
@@ -0,0 +1,177 @@
+        /// <param name="message">The message itself</param>
+        /// <param name="withTimeStamp">true if the message should include a timestamp, false (default) it no timestamp should be included</param>
+        protected void WriteIdentifiedWarning(string cmdletName, string topic, string message, bool withTimeStamp = false)
+        {
+            string formattedMessage = string.Format(
+                CultureInfo.InvariantCulture,
+                "[{0}] {1}: {2}",
+                cmdletName,
+                topic,
+                message);
+
+            if (withTimeStamp)
+            {
+                WriteWarningWithTimestamp(formattedMessage);
+            }
+            else
+            {
+                WriteWarning(formattedMessage);
+            }
+        }
+
+        /// <summary>
+        /// Execute the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            string code = null;
+            HttpStatusCode? statusCode = null;
+            string reasonPhrase = null;
+            string message = null;
+            string exName = null;
+
+            try
+            {
+                this.ProcessRecordInternal();
+            }
+            catch (AggregateException ex)
+            {
+                // Process the exception to be as informative as possible to the user
+                var exTemp = ex.Flatten().InnerException ?? ex;
+                exName = exTemp.GetType().Name;
+                message = exTemp.Message;
+
+                if (exTemp is RestException)
+                {
+                    // Extract relevant information for CloudExceptions and throw
+                    var cloudException = exTemp as CloudException;
+                    if (cloudException != null)
+                    {
+                        message = cloudException.Body.Message;
+                        code = cloudException.Body.Code;
+                        statusCode = cloudException.Response.StatusCode;
+                        reasonPhrase = cloudException.Response.ReasonPhrase;
+                        throw cloudException;
+                    }
+                    else
+                    {                        
+                        // New model to report errors (from Swagger Spec)
+                        var errorResponse = exTemp as ErrorResponseException;
+                        if (errorResponse != null)
+                        {
+                            statusCode = errorResponse.Response.StatusCode;
+                            reasonPhrase = errorResponse.Response.ReasonPhrase;
+                        }
+                        else
+                        {
+                            message = exTemp.ToString();
+                        }
+
+                        throw errorResponse;
+                    }
+                }
+
+                throw exTemp;
+            }
+            catch (Exception ex)
```

Comment:
do not do this elaborate exception catching and rethrowing.  You want to pass exceptions through (especially Aggregate and REST exceptions).  The standard cmdlet menchanisms will handle this in a standard way.

Reply:
Do you mean adding just a simple try catch?
try {
  ...
}
catch (Exception ex){
   throw ex;
}

Reply:
No, I mean removing this code